roachtest: add `failover` chaos tests

erikgrinaker opened the pull request:
**roachtest: export `Failer` in failover tests**

This avoids linters complaining about a `failer` local shadowing the `failer` interface.

**roachtest: add `Failer.Mode()` for failover tests**

**roachtest: reset disk stall detector in `failover/pause`**

The pause failure mode needs to re-enable the disk stall detector after a failure, for use in chaos tests.

**roachtest: allow disabling local SSD via `PreferLocalSSD()`**

**roachtest: add `failover` chaos tests**

This patch adds chaos tests to the `failover` suite. It sets up a 9-node cluster with RF=5 and runs both read-only and read-write workloads against it while failing 1 or 2 random nodes using random failure modes for 1 minute each. As with the other `failover` tests, the pMax unavailability is measured for graphing.

Resolves #103260.
Epic: none
Release note: None

erikgrinaker committed:
roachtest: add `failover` chaos tests

This patch adds chaos tests to the `failover` suite. It sets up a 9-node
cluster with RF=5 and runs both read-only and read-write workloads
against it while failing 1 or 2 random nodes using random failure modes
for 1 minute each. As with the other `failover` tests, the pMax
unavailability is measured for graphing.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/103263)
<!-- Reviewable:end -->

erikgrinaker committed:
roachtest: export `Failer` in `failover` tests

This avoids linters complaining about a `failer` local shadowing the
`failer` interface.

Epic: none
Release note: None

erikgrinaker committed:
roachtest: reset disk stall detector in `failover/pause`

The pause failure mode needs to re-enable the disk stall detector after
a failure, for use in chaos tests.

Epic: none
Release note: None

erikgrinaker committed:
roachtest: add `Failer.Mode()` for `failover` tests

Epic: none
Release note: None

erikgrinaker committed:
roachtest: allow disabling local SSD via `PreferLocalSSD()`

Epic: none
Release note: None

erikgrinaker commented:
Rebased.

andrewbaptist reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>
Thanks. This all looks great. Looking forward to seeing these tests being more reliable and useable in chaos.

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 3 of 4 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/103263#-:-NWO56gOCMVOS0noZ9KH:b-stgoh)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @herkolategan and @srosenberg)



<!-- Sent from Reviewable.io -->

erikgrinaker commented:
TFTR!

bors r+
